service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/FurtherEvidenceUploadedEventHandlerTest.java

```diff
@@ -0,0 +1,320 @@
+import uk.gov.hmcts.reform.fpl.service.FurtherEvidenceNotificationService;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ExtendWith(MockitoExtension.class)
+class FurtherEvidenceUploadedEventHandlerTest {
+    private static final String CASE_REFERENCE = "12345";
+    private static final String CONFIDENTIAL_MARKER = "CONFIDENTIAL";
+    private static final String LA_USER = "LA";
+    private static final String HMCTS_USER = "HMCTS";
+    private static final String REP_USER = "REP";
+    private static final String LA_USER_EMAIL = "la@examaple.com";
+    private static final String HMCTS_USER_EMAIL = "hmcts@examaple.com";
+    private static final String REP_SOLICITOR_USER_EMAIL = "rep@examaple.com";
+    private static final LocalDateTime HEARING_DATE = LocalDateTime.now().plusMonths(3);
+    private static final Long CASE_ID = 12345L;
+    private static final String SENDER_FORENAME = "The";
+    private static final String SENDER_SURNAME = "Sender";
+    private static final String SENDER = SENDER_FORENAME + " " + SENDER_SURNAME;
+
+    @Mock
+    FurtherEvidenceNotificationService furtherEvidenceNotificationService;
+
+    @InjectMocks
+    FurtherEvidenceUploadedEventHandler furtherEvidenceUploadedEventHandler;
+
+    private final Set<String> representativeSolicitors = Set.of("rep_solicitor1@example.com",
+        "rep_solicitor2@example.com",
+        REP_SOLICITOR_USER_EMAIL);
+    private final Set<String> laSolicitors = Set.of(LA_USER_EMAIL);
+
+    @Test
+    void shouldSendNotificationWhenNonConfidentialDocIsUploadedByLA() {
+        CaseData caseData = buildCaseDataWithNonConfidentialLADocuments();
+        FurtherEvidenceUploadedEvent furtherEvidenceUploadedEvent =
+            new FurtherEvidenceUploadedEvent(
+                caseData,
+                buildCaseDataWithConfidentialLADocuments(),
+                true,
+                userDetailsLA()
+            );
+
+        when(furtherEvidenceNotificationService.getRespondentRepresentativeEmails(caseData)).thenReturn(
+            representativeSolicitors);
+
+        furtherEvidenceUploadedEventHandler.handleDocumentUploadedEvent(furtherEvidenceUploadedEvent);
+
+        verify(furtherEvidenceNotificationService).sendFurtherEvidenceDocumentsUploadedNotification(
+            caseData, representativeSolicitors, SENDER);
+    }
+
+    @Test
+    void shouldSendNotificationWhenDocIsMadeNonConfidentialByLA() {
+        CaseData caseDataBefore = buildCaseDataWithConfidentialLADocuments();
+
+        // buildCaseDataWithConfidentialLADocuments() has a "confidential-doc-1" marked as confidential
```

Comment:
tbh I'm not big fan of this, there is too much non explicit data correlation that reader of this tests must know to understand (and need for comment just prove it :))
maybe it would be better
```
        SupportingEvidenceBundle bundle = createBundle();
        SupportingEvidenceBundle updatedBundle = bundle.toBuilder().confidential(emptyList()).build();

        CaseData caseDataBefore = buildCaseDataWithConfidentialLADocuments(bundle);
        CaseData caseData = buildCaseDataWithConfidentialLADocuments(updatedBundle);
```
with this reader immediately see scenario - there is some document, than it is updated to be non confidential and email is send at the end. Reader does not need to know irrelevant details like what is name of doc, can focus on key aspect of test, confidentiality changed, rest is just distraction

But I understand that this may required a lot of additional work, so may stay as it is.